src/system.c

```diff
@@ -1498,6 +1491,41 @@ static void SySetGapRootPath( const Char * string )
     }
 }
 
+/****************************************************************************
+**
+*F  SySetInitialGapRootPaths( <string> )  . . . . .  set the root directories
+**
+**  Set up GAP's initial root paths, based on the location of the
+**  GAP executable.
+*/
+static void SySetInitialGapRootPaths(void)
+{
+    if (GAPExecLocation[0] != 0) {
+        // GAPExecLocation might be a subdirectory of GAP root,
+        // so we will go and search for the true GAP root.
+        // We try stepping back up to two levels.
+        char pathbuf[GAP_PATH_MAX];
+        char initgbuf[GAP_PATH_MAX];
+        strxcpy(pathbuf, GAPExecLocation, sizeof(pathbuf));
+        for (Int i = 0; i < 3; ++i) {
```

Comment:
One more thing, which I asked before: What exactly is our justification for the above, i.e. for looking up to two levels above the `gap` binary for the library? It made sense when we put the executable in `bin/$GAPARCH/`, but we don't anymore...

Also as I wrote before, it turns out that I place my out-of-tree builds into `build/SOME-NAME/`, so for me the above *is* actually useful, but that seems like a somewhat arbitrary justification; still, if that's the reason, I am fine with it; I'd just like to explicitly know what the motivation for the above is.

Reply:
I don't remember you asking before. The motivation is definitely just when this was written we still put the executable in `bin/$GAPARCH/`.

Also, I still tend to still use the executable in bin/$GAPARCH, I didn't realize it was just a symlink.